packages/gestalt/src/Dropdown.js

```diff
@@ -1,73 +1,116 @@
 // @flow strict
-import type { Node } from 'react';
-
-import { Children, cloneElement, useState } from 'react';
+import { Children, cloneElement, type Node, useState } from 'react';
 import PropTypes from 'prop-types';
 import Box from './Box.js';
 import Popover from './Popover.js';
 import Layer from './Layer.js';
 import DropdownItem from './DropdownItem.js';
+import DropdownLink from './DropdownLink.js';
 import DropdownSection from './DropdownSection.js';
 import { DropdownContextProvider } from './DropdownContext.js';
 import { type Indexable, UnsafeIndexablePropType } from './zIndex.js';
 import { type DirectionOptionType } from './utils/keyboardNavigation.js';
-import { type AbstractEventHandler } from './AbstractEventHandler.js';
 import { ESCAPE, SPACE, TAB, ENTER, UP_ARROW, DOWN_ARROW } from './keyCodes.js';
-import { type OptionObject } from './MenuOption.js';
+
+const KEYS = {
+  UP: -1,
+  DOWN: 1,
+  ENTER: 0,
+};
+
+const dropdownItemDisplayNames = ['DropdownItem', 'DropdownLink'];
+
+function getChildrenOptions(childrenArray) {
```

Comment:
Almost nothing changed here, except as noted. Pulled these helper functions out of the component for readability and to make their (lack of) dependencies clear